src/core/qgsogcutils.cpp

```diff
@@ -3374,22 +3374,15 @@ QgsExpressionNodeBinaryOperator *QgsOgcUtilsExpressionFromFilter::nodeBinaryOper
       QString oprValue = static_cast<const QgsExpressionNodeLiteral *>( opRight.get() )->value().toString();
       if ( !wildCard.isEmpty() && wildCard != QLatin1String( "%" ) )
       {
+        // escape default wildcard
         oprValue.replace( '%', QLatin1String( "\\%" ) );
-        if ( oprValue.startsWith( wildCard ) )
+        // split value by
```

Comment:
comment is truncated. 

I also believe your changes has a flow when processing something like "a\\*b" (two backslash characters, not C/Python escaping, so meaning "a, backslash escaped, *, b))  where escapeChar='\' and wildCard='*'
If I follow the code, the oprValue value after this block will be transformed to "a\*b", whereas it should be "a\\%b". There's no testing of this case. So maybe it is not possible to simplify the current code